tests/integration/features.go

```diff
@@ -259,144 +259,319 @@ func addCollectorSpecificMetricLabels(labels sumologicmock.Labels, releaseName s
+	opts := *ctxopts.KubectlOptions(ctx)
+	opts.Namespace = internal.LogsGeneratorNamespace
+	terrak8s.RunKubectl(t, &opts, "delete", "daemonset", internal.LogsGeneratorName)
+	return ctx
+}
+
+func GetAllLogsFeature(waitFunction stepfuncs.WaitForLogs, generate bool) features.Feature {
+	feature := features.New("logs")
+	if generate {
+		feature = feature.
+			Setup(generateDeploymentLogs()).
+			Setup(generateDaemonsetLogs())
+	}
+	feature = feature.
+		Assess("logs from log generator deployment present", waitForDeploymentLogs(
 			logsGeneratorCount,
-			internal.LogsGeneratorName,
-			internal.LogsGeneratorNamespace,
-			internal.LogsGeneratorImage,
+			false,
+			waitFunction,
 		)).
-		Setup(stepfuncs.GenerateLogs(
-			stepfuncs.LogsGeneratorDaemonSet,
+		Assess("logs from log generator daemonset present", waitForDaemonsetLogs(
 			logsGeneratorCount,
-			internal.LogsGeneratorName,
-			internal.LogsGeneratorNamespace,
-			internal.LogsGeneratorImage,
+			false,
+			waitFunction,
 		)).
-		Assess("logs from log generator deployment present", stepfuncs.WaitUntilExpectedLogsPresent(
+		Assess("expected container log metadata is present for log generator deployment", checkDeploymentMetadata(
 			logsGeneratorCount,
-			map[string]string{
-				"namespace":      internal.LogsGeneratorName,
-				"pod_labels_app": internal.LogsGeneratorName,
-				"deployment":     internal.LogsGeneratorName,
-			},
-			waitDuration,
-			tickDuration,
+			false,
+			waitFunction,
 		)).
-		Assess("logs from log generator daemonset present", stepfuncs.WaitUntilExpectedLogsPresent(
+		Assess("expected container log metadata is present for log generator daemonset", checkDaemonsetMetadata(
 			logsGeneratorCount,
-			map[string]string{
-				"namespace":      internal.LogsGeneratorName,
-				"pod_labels_app": internal.LogsGeneratorName,
-				"daemonset":      internal.LogsGeneratorName,
-			},
-			waitDuration,
-			tickDuration,
+			false,
+			waitFunction,
+		)).
+		Assess("logs from node systemd present", checkSystemdLogs(
+			10, // we don't really control this, just want to check if the logs show up
+			false,
+			waitFunction,
+		)).
+		Assess("logs from kubelet present", checkKubeletLogs(
+			1, // we don't really control this, just want to check if the logs show up
+			false,
+			waitFunction,
+		))
+
+	if generate {
+		feature = feature.
+			Teardown(removeLogsDeployment).
+			Teardown(removeLogsDaemonset).
+			Teardown(stepfuncs.KubectlDeleteNamespaceOpt(internal.LogsGeneratorNamespace))
+	}
+
+	return feature.Feature()
+}
+
+func DeployAdditionalSumologicMock() features.Feature {
+	return features.New("create additional sumologic mock").
+		Setup(stepfuncs.KubectlApplyFOpt(internal.YamlPathAdditionalSumologicMock, internal.AdditionalSumologicMockNamespace)).
+		Assess("additional sumologic mock is ready", stepfuncs.WaitUntilAdditionalSumologicMockAvailable(waitDuration, tickDuration)).
+		Feature()
```

Comment:
We should probably create the namespace here if needed and put it in kubectlOptions, and then in the deletion function below we can unset and delete it.

On that note, would it be a problem to make this namespace name depend on the test name? Otherwise it's annoying to run multiple tests in the same cluster, as namespaces take a while to delete.

Reply:
I need to test it, but it should be possible

Reply:
It would require generating configuration for test in-fly. It requires more changes which I want to include in this PR